### PR TITLE
Default config updates to recognize more defaults

### PR DESCRIPTION
**Pull request by onovotny**
This addresses #620

### CODE REVIEW

**Comment by JakeGinnivan** [comment]
We need to do a config upgrade for all these.

If someone has provided config overrides for these different branch configurations we can't have the new configs taking over

File: src/GitVersionCore/Configuration/ConfigurationProvider.cs [hunk 1]
```diff
@@ -31,15 +31,15 @@ public static void ApplyDefaultsTo(Config config)
             var configBranches = config.Branches.ToList();
 
             ApplyBranchDefaults(config, GetOrCreateBranchDefaults(config, "master"), defaultTag: string.Empty, defaultPreventIncrement: true);
-            ApplyBranchDefaults(config, GetOrCreateBranchDefaults(config, "release[/-]"), defaultTag: "beta", defaultPreventIncrement: true);
```
Comment (JakeGinnivan) [inline src/GitVersionCore/Configuration/ConfigurationProvider.cs]: Add another parameter to this which is `migrate:` and then it takes an array. You can then add `migrate: new[] { "release[/-]" }` and ApplyBranchDefaults should rename the configs to be migrated, then apply the defaults.

Just one idea on how to solve.

Comment (onovotny) [inline src/GitVersionCore/Configuration/ConfigurationProvider.cs]: I guess I'm not sure how the migration should work? Should apply branch defaults look for the old keys and rename them?

Comment (JakeGinnivan) [inline src/GitVersionCore/Configuration/ConfigurationProvider.cs]: Yeah exactly. So I have:

```
branches:
  develop:
    tag: beta
```

The effective config will have both `develop` and `dev(elop)?...`. We need to look for develop, rename it to the new one then delete it

**Comment by onovotny** [comment]
Okay this should be working now. Also, the CanReadDocument test is now CanReadDocumentAndMigrate as that proves that it works.

**Pull request merged**